Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -482,32 +489,32 @@ void HUDController::AbilityCoolDownEffectCheck(Cooldowns cooldown, GameObject* c
 
 				if (cooldown < Cooldowns::ONIMARU_SKILL_1) {
 					if (canvas->GetChildren().size() > 0) {
-						if (canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren().size() > 5) {
-							ef = GET_SCRIPT(canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren()[0], AbilityRefreshEffect);
+						if (canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren().size() > HIERARCHY_INDEX_MAIN_BUTTON_DOWN) {
+							ef = GET_SCRIPT(canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren()[HIERARCHY_INDEX_MAIN_ABILITY_EFFECT], AbilityRefreshEffect);
 							//Turn on button idle
-							canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren()[5]->Enable();
+							canvas->GetChildren()[static_cast<int>(cooldown)]->GetChild(HIERARCHY_INDEX_MAIN_BUTTON_UP)->Enable();
 							//Turn off button down
-							canvas->GetChildren()[static_cast<int>(cooldown)]->GetChildren()[6]->Disable();
+							canvas->GetChildren()[static_cast<int>(cooldown)]->GetChild(HIERARCHY_INDEX_MAIN_BUTTON_DOWN)->Disable();
 						}
 					}
 				} else if (cooldown < Cooldowns::SWITCH_SKILL) {
 					if (canvas->GetChildren().size() > 0) {
-						if (canvas->GetChildren()[static_cast<int>(cooldown) - 3]->GetChildren().size() > 5) {
-							ef = GET_SCRIPT(canvas->GetChildren()[static_cast<int>(cooldown) - 3]->GetChildren()[0], AbilityRefreshEffect);
+						if (canvas->GetChildren()[static_cast<int>(cooldown) - 3]->GetChildren().size() > HIERARCHY_INDEX_MAIN_BUTTON_DOWN) {
```

Comment:
Main button down is 6 and the old value is 5. Is this correct?

Reply:
Nicely spotted